### ISSUE TITLE
Append branch fails when adding a branch that violates unique node alias constraint

### ISSUE THREAD

**Issue by robgaston** [archesproject/arches#8878]
Appending a branch that violates the unique node alias constraint on the target resource model results in an exception that is not easy to understand.

We should probably simply resolve the error by adjusting the node aliases on append.

**Comment by Habennin** [comment]
This is a smart solution. The way that we use branches at the Getty is to create really common shapes that are reused across branches so we might typically use the same branch 3 or more times in the same model, in different places.

We have a naming scheme where we have a regimented way of naming the branches as they are appended into the models anyhow. This involves prepending the name of the semantic node into which they are pasted. It would be fantastic if as you added the branch you were asked for a prepend name that it would add to the branches as they were added in. This would save A LOT of work.

**Comment by azaroth42** [comment]
+1 - this is costing me a LOT of time at the moment :(

**Comment by chiatt** [comment]
Fixed in #9209 and #9210

**Comment by azaroth42** [comment]
Running latest, this still doesn't work :(

**Comment by azaroth42** [comment]
If the nodes are only at a single level, then it works as expected. However when grafting a deeply nested branch on to a model, it doesn't find all of the nodes and hence ends up with a duplicate name.

**Comment by chrabyrd** [comment]
This could apply, it's in review and hasn't been merged yet:

https://github.com/archesproject/arches/pull/9473

**Comment by azaroth42** [comment]
That looks promising!

My current work around is:

```
    def make_name_unique(self, name, names_to_check, suffix_delimiter="_"):
        """
        Makes a name unique among a list of names

        Arguments:
        name -- the name to check and modfiy to make unique in the list of "names_to_check"
        names_to_check -- a list of names that "name" should be unique among
        """

        i = random.randint(len(names_to_check), len(names_to_check)+1000)  
        temp_node_name = "{0}{1}{2}".format(name, suffix_delimiter, i)
        return temp_node_name
```

**Comment by jacobtylerwalls** [comment]
Fixed in #9473, but correct me if not the case.
